.github/workflows/release-and-publish.yaml

```diff
@@ -1,93 +1,72 @@
 name: Release and publish
 
 on:
-  workflow_dispatch:
-    inputs:
-      release-type:
-        type: choice
-        description: |
-          Release type
-          Use prepatch, preminor, premajor for the first "pre", to increment pre value use prerelease.
-        options:
-          - prerelease
-          - patch
-          - minor
-          - major
-          - prepatch
-          - preminor
-          - premajor
-        required: true
+  push:
+    branches:
+      - "main"
+
 jobs:
   build:
+    if: ${{startsWith(github.event.head_commit.message, 'Release *')}}
     runs-on: ubuntu-latest
     steps:
+      - name: get version from commit
+        env:
+          C_MESSAGE: ${{github.event.head_commit.message}}
+        # Commit message following structure: "Release *v1.0.3*", "Release *v2.0.0-beta.2"
+        run: |
+          echo "NEW_VERSION=$(echo $C_MESSAGE | cut -d'*' -f 2)" >> $GITHUB_ENV
+      - name: Verify version regex
+        if: ${{ contains(env.NEW_VERSION, 'beta') != true }}
+        run: |
+          [[ ! $NEW_VERSION =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]] && echo "Version '$NEW_VERSION' does not match regex" && exit 1
+      - name: Verify beta version regex
+        if: ${{ contains(env.NEW_VERSION, 'beta') == true }}
+        run: |
+          [[ ! $NEW_VERSION =~ ^v[0-9]+\.[0-9]+\.[0-9]+-beta\.[0-9]+$ ]] && echo "Version '$NEW_VERSION' does not match regex" && exit 1
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: "16.x"
+          node-version: "18.x"
```

Comment:
Maybe add npm caching to speed up builds, iff it makes sense. 

https://github.com/actions/setup-node#caching-global-packages-data